src/routes/_tabLayout/flash-node.lazy.tsx

```diff
@@ -1,42 +1,48 @@
 import { createLazyFileRoute } from "@tanstack/react-router";
 import type { AxiosProgressEvent } from "axios";
 import { filesize } from "filesize";
-import { useEffect, useRef, useState } from "react";
+import { Cpu } from "lucide-react";
+import { useEffect, useMemo, useRef, useState } from "react";
 
 import ConfirmationModal from "@/components/ConfirmationModal";
 import TabView from "@/components/TabView";
 import { Button } from "@/components/ui/button";
 import { Checkbox } from "@/components/ui/checkbox";
 import { Input } from "@/components/ui/input";
 import { Progress } from "@/components/ui/progress";
-import {
-  Select,
-  SelectContent,
-  SelectItem,
-  SelectTrigger,
-  SelectValue,
-} from "@/components/ui/select";
+import { Toggle } from "@/components/ui/toggle";
 import { useToast } from "@/hooks/use-toast";
 import { useNodeUpdateMutation } from "@/lib/api/file";
-import { useFlashStatusQuery } from "@/lib/api/get";
+import { useAboutTabData, useFlashStatusQuery } from "@/lib/api/get";
 
 export const Route = createLazyFileRoute("/_tabLayout/flash-node")({
   component: Flash,
 });
 
-interface SelectOption {
-  value: string;
-  label: string;
-}
-
-const nodeOptions: SelectOption[] = [
-  { value: "0", label: "Node 1" },
-  { value: "1", label: "Node 2" },
-  { value: "2", label: "Node 3" },
-  { value: "3", label: "Node 4" },
+const nodeOptions = [
+  { value: 0, label: "Node 1" },
+  { value: 1, label: "Node 2" },
+  { value: 2, label: "Node 3" },
+  { value: 3, label: "Node 4" },
 ];
 
+const isSemverGreaterOrEqual = (a: string, b: string) => {
+  return a.localeCompare(b, undefined, { numeric: true }) >= 0;
+};
+
 function Flash() {
+  const { data: aboutData } = useAboutTabData();
+  const batchFlashingSupport = useMemo(
+    () => isSemverGreaterOrEqual(aboutData.board_revision, "2.5"),
```

Comment:
Here, I'm assuming the new hardware feature is incremental (e.g., the Turing 2.6, 2,7, and so on will also have batch flashing). It's not something to worry about right now, but if that's not the case in the future, it may need revision.

Reply:
you are right